src/calling/haplotypes.rs

```diff
@@ -34,11 +34,7 @@ impl Caller {
 
         // Step 3: calculate posteriors.
         //let m = model.compute(universe, &data);
-        //let n_of_haplotypes = 3;
-        let mut haplotypes = Vec::new();
-        for key in kallisto_estimates.keys() {
-            haplotypes.push(key);
-        }
+        let haplotypes: Vec<_> = kallisto_estimates.keys().map(|x| x.to_string()).collect();
```

Comment:
Instead of the map(), you can use cloned(), which has the same effect. No need to write the map down manually.

Reply:
But the extend method is only accepting the same types to be extended. That's why I needed to have the `to_string`.